test/test_cuda.py

```diff
@@ -1571,6 +1571,10 @@ def test_mean_fp16(self):
         x = torch.ones(65536, device='cuda', dtype=torch.float16)
         self.assertEqual(x.mean(dtype=torch.float32), 1)
 
+    def test_prod_large(self):
+        x = torch.ones(240000, device='cuda', dtype=torch.float32)
+        self.assertEqual(x.prod(), 1)
```

Comment:
Please also add a comment explaining why do we need to test this. Is it that we use different kernels for large vs small inputs? Those things are super helpful when refactoring the tests.

Reply:
There is now a single kernel for all reductions, with the different options. I'm not sure all the possible options are thoroughly tested, it would be a good idea to test them, but I don't know exactly in which cases which options are triggered.